src/ProvidedTypes.fs

```diff
@@ -8722,7 +8722,7 @@ namespace ProviderImplementation.ProvidedTypes
                     convTypeDefToTgt t
                     
                 | _ -> 
-                let asms = (if toTgt then getTargetAssemblies() else getSourceAssemblies())
+                let asms = System.Linq.Enumerable.Reverse(if toTgt then getTargetAssemblies() else getSourceAssemblies())

```

Comment:
This is expensive, reversing the thing every time.  
So the idea is good but we should adjust

Reply:
How about just a index based loop on the ResizeArray?